### ISSUE TITLE
Error: The migration was modified after it was applied always shows up after clone the project from git

### ISSUE THREAD

**Issue by zhangyi921** [prisma/prisma#8123]
### Bug description

I have one prisma repository and cloned it into my computer and everything works fine. If I run 
```npx prisma migrate dev``` twice and the second time it will show ```Already in sync, no schema change or pending migration was found.``` which is great.

Then my teammate clone the project and install all the dependencies and run ```npx prisma migrate dev```, then it shows ```The migration `...` was modified after it was applied.``` and database reset is required. At this point no one has modified schema file nor database, and there is no data in the database. 

If my teammate reset the database, I will have the "The migration was modified" error when I run ```npx prisma migrate dev``` even though schema is not changed.  Basically everyone in the team needs to reset the database when they clone or some one else changes the schema.

### How to reproduce

1. Create a prisma project
2. Add a model in the schema file and apply it to database
3. Clone the project into a different directory
4. Run ```npx prisma migrate dev```
5. Error "The migration was modified" shows up.

### Expected behavior

After apply the schema to the database, clone the prisma project again(schema not changed), run ```npx prisma migrate dev```. It should show ```Already in sync, no schema change or pending migration was found.``` instead of ```The migration `...` was modified after it was applied.```

### Prisma information

generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model Organization {
  id              Int            @id @default(autoincrement())
  createdAt       DateTime       @default(now())
  name            String         @unique

}

### Environment & setup

- OS: Windows 10
- Database: [PostgreSQL]
- Node.js version: v14.17.1

### Prisma Version

```
Environment variables loaded from .env
prisma               : 2.26.0
@prisma/client       : 2.26.0
Current platform     : windows
Query Engine         : query-engine 9b816b3aa13cc270074f172f30d6eda8a8ce867d (at node_modules\@prisma\engines\query-engine-windows.exe)
Migration Engine     : migration-engine-cli 9b816b3aa13cc270074f172f30d6eda8a8ce867d (at node_modules\@prisma\engines\migration-engine-windows.exe)
Introspection Engine : introspection-core 9b816b3aa13cc270074f172f30d6eda8a8ce867d (at node_modules\@prisma\engines\introspection-engine-windows.exe)
Format Binary        : prisma-fmt 9b816b3aa13cc270074f172f30d6eda8a8ce867d (at node_modules\@prisma\engines\prisma-fmt-windows.exe)
Default Engines Hash : 9b816b3aa13cc270074f172f30d6eda8a8ce867d
Studio               : 0.408.0
```

**Comment by tomhoule** [comment]
Is your colleague on OSX/Linux? There is a known problem there because git will automatically change the line endings to `\r\n` when checking out on windows and to `\n` on Unix-like systems. So migrate actually sees different files. It is probably something that can be fixed, thanks for reporting the issue.

**Comment by zhangyi921** [comment]
We are all on Windows 10. Even I clone the project into my computer's two different directories, I still can reproduce the error.

**Comment by tomhoule** [comment]
Ok that could be a different issue then. We need to look into this.

**Comment by zhangyi921** [comment]
I find a temporary fix: copy migration files from a working repo to a not working repo, then the error disappears and both start working. So it is a line ending bug.

**Comment by thebiglabasky** [comment]
We'll wait for @tomhoule to come back from vacation to rediscuss this one

**Comment by SCLeoX** [comment]
I encountered a similar issue where git changed all LF line endings to CRLF line endings, causing the production server to throw a bunch of errors regarding migration being modified. I think Prisma should internally convert all CRLF's to LF's before calculating the checksum. Or, to be more backwards compatible, maybe try to do that if the checksum mismatched.

**Comment by thebiglabasky** [comment]
We recently fixed the end-of-line problem in `2.29.0`.
Could you try again and see if it helped?
Thank you

**Comment by tomhoule** [comment]
Since we are confident this is fixed from reports from other users, let's close this issue. Please open a new one if something similar happens.
